tools/azure-sdk-tools/devtools_testutils/resource_testcase.py

```diff
@@ -50,7 +51,7 @@ def __init__(self, name_prefix='',
             self._need_creation = False
         if self.random_name_enabled:
             self.resource_moniker = self.name_prefix + "rgname"
-        self.set_cache(use_cache, parameter_name)
+        self.set_cache(use_cache, parameter_name, name_prefix)
```

Comment:
I'm wondering if we still need this or if it was a spurious part of our debugging.  If we can do without I'd like to do without, since it's basically saying "the name prefix is integral to this resource being The Same Resource" (e.g. that tests rely on fixed name prefixes) which is a bad smell, but not the end of the world.